arbor/include/arbor/common_types.hpp

```diff
@@ -138,8 +138,8 @@ using time_type = double;
 constexpr time_type terminal_time = std::numeric_limits<time_type>::max();
 
 // For holding counts and indexes into generated sample data.
-
-using sample_size_type = std::int32_t;
+using sample_index_type = std::int32_t;
```

Comment:
where is this type actually used?

Reply:
It -- currently -- isn't, but it's useful when we are considering things like:
```cxx
sample_size_type a = ..., b = ..., c = a - b;
```
due to the potential overflow.